networkx/algorithms/community/demon.py

```diff
@@ -0,0 +1,296 @@
+# -*- coding: utf-8 -*-
+#    Copyright (C) 2015
+#    All rights reserved.
+#    BSD license.
+
+"""
+Flat Merge version of DEMON (Democratic Estimate of the Modular Organization of
+a Network) algorithm as described in:
+
+Michele Coscia, Giulio Rossetti, Fosca Giannotti, Dino Pedreschi:
```

Comment:
This is user-facing documentation, don't have raw references like this. The reference should live in the docstring for the particular function that implements the algorithm. The module-level documentation should describe at a high-level what functions are provided by this module.